cms/djangoapps/contentstore/utils.py

```diff
@@ -1847,7 +1872,10 @@ def get_container_handler_context(request, usage_key, course, xblock):  # pylint
         create_xblock_info,
     )
     from openedx.core.djangoapps.content_staging import api as content_staging_api
+    from cms.djangoapps.contentstore.views.course import _course_outline_json
 
+    course_structure = _course_outline_json(request, course)
```

Comment:
Instead of creating JSON representation of a course, [with all additional keys for each xblock](https://github.com/raccoongang/edx-platform/blob/dd1f2081994e900db3a85d3c7114ff0a745e4fd6/cms/djangoapps/contentstore/xblock_storage_handlers/view_handlers.py#L1241), we should work directly with `course` object to save computational resources and make it faster.

Please see example for iterating over the course object (Sections) https://github.com/openedx/edx-platform/blob/bca34c0993aa7004b86cbf318ea55fd3a0ca5d95/cms/djangoapps/contentstore/utils.py#L1005, so you can collect subsections IDs